### PR TITLE
Fix warnings in StatementResource

### PR DESCRIPTION
**Pull request by electrum**

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/server/StatementResource.java [hunk 1]
```diff
@@ -573,11 +573,11 @@ private synchronized URI createNextResultsUri(UriInfo uriInfo)
 
         private static List<Column> createColumnsList(QueryInfo queryInfo)
         {
-            requireNonNull(queryInfo, "queryInfo is null");
```
Comment (dain) [inline presto-main/src/main/java/com/facebook/presto/server/StatementResource.java]: Did you mean to remove the null check?

File: presto-main/src/main/java/com/facebook/presto/server/StatementResource.java [hunk 2]
```diff
@@ -573,11 +573,11 @@ private synchronized URI createNextResultsUri(UriInfo uriInfo)
...
-            checkArgument(queryInfo.getOutputStage().isPresent(), "outputStage is not present");
+            StageInfo outputStage = queryInfo.getOutputStage()
+                    .orElseThrow(() -> new IllegalArgumentException("outputStage not present"));
```
Comment (dain) [inline presto-main/src/main/java/com/facebook/presto/server/StatementResource.java]: I like this better the other way

**Comment by dain** [comment]
One comment. Otherwise looks good.

File: presto-main/src/main/java/com/facebook/presto/server/StatementResource.java (hunk 1 above)
Comment (electrum) [inline presto-main/src/main/java/com/facebook/presto/server/StatementResource.java]: There's no need for it, this is a private method that is always called with non-null

File: presto-main/src/main/java/com/facebook/presto/server/StatementResource.java (hunk 2 above)
Comment (dain) [inline presto-main/src/main/java/com/facebook/presto/server/StatementResource.java]: ah ok.  Never mind then

Comment (electrum) [inline presto-main/src/main/java/com/facebook/presto/server/StatementResource.java]: IDEA warns that `get()` is called without a check. It doesn't like things like this:
```java
checkArgument(object.getOptionalField().isPresent());
object.getOptionalField.get();
```
Arguably, it should do a better job for getters that return a final field.

**Pull request closed without merging**